### PR TITLE
Feature/cookbook recipes with ings

### PR DESCRIPTION
**Pull request by aliroemhildt**
#### What's this PR do?
highlights recipe cards on cookbook page if a user has enough ingredients to cook that recipe

#### Where should the reviewer start?
scripts at line 144

#### How should this be manually tested?
- hard to test on site because we're not sure if any users actually have enough ingredients to cook any of the recipes
- we will need to come back to this and figure out how to test to make sure it's working

#### Any background context you want to provide?
- the css for the highlight key needs a little work, I can play around with it

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -126,9 +126,12 @@ <h3>imma lookin'</h3>
         </div>
         <button id="filterButton">apply filter</button>
       </section>
-      <section class="recipes-section recipes-section-js">
+      <section class="recipes-wrapper">
+        <p class="key hidden">Yellow Highlight: You have enough ingredients to cook this recipe!</p>
+        <section class="recipes-section recipes-section-js">
```
Comment (cagallo) [inline src/index.html:131]: This is a great idea! Great way to indicate which recipes the user can cook.

**Review by kammon10 — APPROVED** [review]
beautiful Ali!

File: src/index.html [hunk 2]
```diff
@@ -126,9 +126,12 @@ <h3>imma lookin'</h3>
         </div>
         <button id="filterButton">apply filter</button>
       </section>
-      <section class="recipes-section recipes-section-js">
+      <section class="recipes-wrapper">
+        <p class="key hidden">Yellow Highlight: You have enough ingredients to cook this recipe!</p>
```
Comment (anewb87) [inline src/index.html:130]: I really like this idea. It immediately shows the user what recipes they can cook based on what's currently in their pantry.

File: src/scripts.js [hunk 3]
```diff
@@ -142,6 +142,23 @@ function displayCookbook() {
...
+      return recipe.id === parseInt(recipeCard.id.slice(2))
+    })
+
+    const haveIngredients = currentUser.pantry.checkPantry(recipeOnCard);
+    if (haveIngredients) {
+      recipeCard.childNodes[3].classList.add('highlight-card');
+    }
+  })
+
+  return recipesUserCanCook;
```
Comment (cagallo) [inline src/scripts.js:161]: This looks great, can't wait to do more testing with it once we can add ingredients to the api!

**Review by cagallo — APPROVED** [review]

File: src/styles.css [hunk 4]
```diff
@@ -34,7 +34,7 @@ button {
 }
 
 button:hover {
-  background: #FCE49E;
```
Comment (anewb87) [inline src/styles.css:37]: Maybe not as 'pretty' a yellow with our other colors, but we believe this is a higher contrast and is better for accessibility.

File: src/styles.css [hunk 5]
```diff
@@ -235,6 +241,18 @@ nav {
...
 
+.key {
+  margin: 18px 0px 0px 18px;
+  font-size: 20px;
+  padding: 8px;
+  background: #E7C423;
+  width: 100%;
+}
+
+.highlight-card {
```
Comment (anewb87) [inline src/styles.css:252]: LOVE! Simple but does the job- signifies which recipes the user can cook based on what's in their panty.

File: src/scripts.js [hunk 6]
```diff
@@ -142,6 +142,23 @@ function displayCookbook() {
...
+}
+
+function showRecipesUserCanCook() {
+  const recipeCards = document.querySelectorAll('.recipe-card-js');
+  const recipesUserCanCook = recipeCards.forEach(recipeCard => {
+    const recipeOnCard = currentUser.recipesToCook.find(recipe => {
+      return recipe.id === parseInt(recipeCard.id.slice(2))
+    })
+
+    const haveIngredients = currentUser.pantry.checkPantry(recipeOnCard);
```
Comment (anewb87) [inline src/scripts.js:155]: Yay for methods in our classes!

**Review by anewb87 — APPROVED** [review]
Moving along.

**Pull request merged**